examples/kotlin/src/main/kotlin/com/example/authors/QueriesImpl.kt

```diff
@@ -0,0 +1,109 @@
+// Code generated by sqlc. DO NOT EDIT.
+
+package com.example.authors
+
+import java.sql.Connection
+import java.sql.SQLException
+
+const val createAuthor = """-- name: createAuthor :one
+INSERT INTO authors (
+          name, bio
+) VALUES (
+  ?, ?
+)
+RETURNING id, name, bio
+"""
+
+data class CreateAuthorParams (
+  val name: String,
+  val bio: String?
+)
+
+const val deleteAuthor = """-- name: deleteAuthor :exec
+DELETE FROM authors
+WHERE id = ?
+"""
+
+const val getAuthor = """-- name: getAuthor :one
+SELECT id, name, bio FROM authors
+WHERE id = ? LIMIT 1
+"""
+
+const val listAuthors = """-- name: listAuthors :many
+SELECT id, name, bio FROM authors
+ORDER BY name
+"""
+
+class QueriesImpl(private val conn: Connection) {
+
+  @Throws(SQLException::class)
+  fun createAuthor(arg: CreateAuthorParams): Author {
+    val stmt = conn.prepareStatement(createAuthor)
+    stmt.setString(1, arg.name)
+    stmt.setString(2, arg.bio)
+
+    return stmt.executeQuery().use { results ->
+      if (!results.next()) {
+        throw SQLException("no rows in result set")
+      }
+      val ret = Author(
+      results.getLong(1),
```

Comment:
you can select by field name, which is slightly nicer and less brittle. 

```
results.getLong("id")
```

Reply:
The generator will ensure the columns match. This won't be brittle.

Reply:
I think there is also a slight performance cost with named columns but I might be wrong.

Reply:
What does sqlc do if you use `*` in your SELECT?

Reply:
The sqlc parser expands the `*` to the actual columns in your schema and rewrites your query.

Reply:
oh wow